src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/MemoryMarshal.cs

```diff
@@ -211,27 +211,45 @@ public static ReadOnlySpan<byte> AsBytes<T>(ReadOnlySpan<T> span)
         }
 
         /// <summary>
-        /// Create a new span over a portion of a regular managed object. This can be useful
+        /// Creates a new span over a portion of a regular managed object. This can be useful
         /// if part of a managed object represents a "fixed array." This is dangerous because the
         /// <paramref name="length"/> is not checked.
         /// </summary>
         /// <param name="reference">A reference to data.</param>
         /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
-        /// <returns>The lifetime of the returned span will not be validated for safety by span-aware languages.</returns>
+        /// <returns>A span representing the specified reference and length.</returns>
+        /// <remarks>The lifetime of the returned span will not be validated for safety by span-aware languages.</remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Span<T> CreateSpan<T>(ref T reference, int length) => new Span<T>(ref reference, length);
 
         /// <summary>
-        /// Create a new read-only span over a portion of a regular managed object. This can be useful
+        /// Creates a new read-only span over a portion of a regular managed object. This can be useful
         /// if part of a managed object represents a "fixed array." This is dangerous because the
         /// <paramref name="length"/> is not checked.
         /// </summary>
         /// <param name="reference">A reference to data.</param>
         /// <param name="length">The number of <typeparamref name="T"/> elements the memory contains.</param>
-        /// <returns>The lifetime of the returned span will not be validated for safety by span-aware languages.</returns>
+        /// <returns>A read-only span representing the specified reference and length.</returns>
+        /// <remarks>The lifetime of the returned span will not be validated for safety by span-aware languages.</remarks>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ReadOnlySpan<T> CreateReadOnlySpan<T>(ref T reference, int length) => new ReadOnlySpan<T>(ref reference, length);
 
+        /// <summary>Creates a new read-only span for a null-terminated string.</summary>
+        /// <param name="value">The pointer to the null-terminated string of characters.</param>
+        /// <returns>A read-only span representing the specified null-terminated string.</returns>
+        /// <remarks>The returned span does not include the null terminator.</remarks>
```

Comment:
What's the behavior if `value` is null OR if `value` doesn't have a null terminator? Might be worth documenting these edge cases as remarks.

If there is no null terminator, I am guessing we throw argument exception (because of `string.strlen`). We should document that exception and add a test case.

Reply:
> if value doesn't have a null terminator?

This is all interop. If you give it garbage native data, that's on you, and behavior is undefined, just like with strlen.  Most likely it'll crash.

>  if value is null

We could catch this and throw for it or return a default span.  Right now it'll null ref.  I don't have a strong opinion.